### ISSUE TITLE
Can I deactivate used env if I cd to folder

### ISSUE THREAD

**Issue by enagorny** [hyperupcall/autoenv#19]
Could current activated env be deactivated if I cd to folder which is not subfolder of current directory?

Also if I using such .env:
    source oenv/bin/activate
it works when I enter to this folder, but then when I go to subfolder I'm getting:
    bash: oenv/bin/activate: No such file or directory
I can fix it using absolute path, but it will be activated twice, isn't it.

**Comment by hwangmoretime** [comment]
+1

**Comment by zambon** [comment]
:+1: for resetting environment variables.

@kennethreitz Limiting the scope for environment variables only (not any bash script) sounds good, but dramatically changes the functionality of `autoenv`, right?

**Comment by kennethreitz** [comment]
Correct.

**Comment by robcowie** [comment]
How about calling a 'tear down' function if it is defined in .env? For example, to deactivate a conda environment...

```shell
on_leave() {
    echo "Deactivating environment"
    source deactivate
}
```

**Comment by HSin-Tse** [comment]
i mkdir "dac" a directory  and touch .env with the line in the "dac"

deactivate
cd ..

######
now i can cd dac and deactivate and back

**Comment by hyperupcall** [comment]
Closing as this has been implemented in [v0.3.0](https://github.com/hyperupcall/autoenv/releases/tag/v0.3.0) with the `.env.leave` file. It's not perfect (autoenv is meant to be as simple as possible), but it's what you would use to do this.
